lib/ui/bottom-panel.js

```diff
@@ -8,65 +8,101 @@ export class BottomPanel {
     this.visibility = false
+    this.visibleMessages = 0
+    this.alwaysTakeMinimumSpace = atom.config.get('linter.alwaysTakeMinimumSpace')
+    this.errorPanelHeight = atom.config.get('linter.errorPanelHeight')
+    this.configVisibility = atom.config.get('linter.showErrorPanel')
     this.scope = scope
     this.messages = new Map()
 
-    this.subscriptions.add(atom.config.observe('linter.showErrorPanel', value => {
-      this.configVisibility = value
-      this.setVisibility(true)
+    // Keep messages contained to measure height.
+    this.element.appendChild(this.messagesElement)
+
+    this.subscriptions.add(atom.config.onDidChange('linter.alwaysTakeMinimumSpace', ({newValue, oldValue}) => {
+      this.alwaysTakeMinimumSpace = newValue
+      this.updateHeight()
+    }))
+
+    this.subscriptions.add(atom.config.onDidChange('linter.errorPanelHeight', ({newValue, oldValue}) => {
+      this.errorPanelHeight = newValue
+      this.updateHeight()
+    }))
+
+    this.subscriptions.add(atom.config.onDidChange('linter.showErrorPanel', ({newValue, oldValue}) => {
+      this.configVisibility = newValue
+      this.updateVisibility()
     }))
+
     this.subscriptions.add(atom.workspace.observeActivePaneItem(paneItem => {
       this.paneVisibility = paneItem === atom.workspace.getActiveTextEditor()
-      this.setVisibility(true)
+      this.updateVisibility()
     }))
 
-    Interact(this.element).resizable({edges: { top: true }})
-      .on('resizemove', function (event) {
-        var target = event.target;
-        if (event.rect.height < 25) {
-          if (event.rect.height < 1) {
-            target.style.width = target.style.height = null
-          } else return // No-Op
-        } else {
-          target.style.width  = event.rect.width + 'px'
-          target.style.height = event.rect.height + 'px'
-        }
+    Interact(this.element).resizable({edges: {top: true}})
+      .on('resizemove', event => {
+        event.target.style.height = `${event.rect.height}px`
+      })
+      .on('resizeend', event => {
+        atom.config.set('linter.errorPanelHeight', event.target.clientHeight)
       })
   }
   refresh(scope) {
     this.scope = scope
+    this.visibleMessages = 0
+
     for (let message of this.messages) {
-      message[1].updateVisibility(scope)
+      const visible = message[1].updateVisibility(scope)
+      if (visible) this.visibleMessages++
     }
+
+    this.updateVisibility()
   }
   setMessages({added, removed}) {
     if (removed.length)
       this.removeMessages(removed)
+
     for (let message of added) {
       const messageElement = Message.fromMessage(message)
-      this.element.appendChild(messageElement)
-      messageElement.updateVisibility(this.scope)
+      this.messagesElement.appendChild(messageElement)
+      const visible = messageElement.updateVisibility(this.scope)
       this.messages.set(message, messageElement)
+      if (visible) this.visibleMessages++
```

Comment:
The same https://github.com/atom-community/linter/pull/929/files#r41109890 applies here